source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOResourceResolvers.mm

```diff
@@ -10,25 +10,43 @@
 @implementation ACOResourceResolvers
 {
     NSMutableDictionary<NSString *, NSObject<ACOIResourceResolver> *> *_resolvers;
+    NSMutableDictionary<NSString *, NSNumber *> *_resolversIFMap;
 }
 
 - (instancetype)init
 {
     self = [super init];
     if (self) {
         _resolvers = [[NSMutableDictionary alloc] init];
+        _resolversIFMap = [[NSMutableDictionary alloc] init];
     }
     return self;
 }
 
 - (void)setResourceResolver:(NSObject<ACOIResourceResolver> *)resolver scheme:(NSString *)scheme
 {
     self->_resolvers[scheme] = resolver;
+
+    _resolversIFMap[scheme] = [NSNumber numberWithInt:ACODefaultIF];
+
+    if([resolver respondsToSelector:@selector(resolveImageResource:)]) {
+        _resolversIFMap[scheme] = [NSNumber numberWithInt:ACOImageIF];
+    }
+    // only one IF per scheme is supported and ACRImageViewIF will be chosen
+    // when both are implemented
+    if([resolver respondsToSelector:@selector(resolveImageViewResource:)]) {
+        _resolversIFMap[scheme] = [NSNumber numberWithInt:ACOImageViewIF];
+    }
 }
```

Comment:
This should be above and have an else.

If (supports resolveImageView)
{
    set ImageView
}
else if (supports resolveImage)
{
    set Image
}

This makes it easier to extend. First priority as the first statement